src/server/scripts/Northrend/Ulduar/Ulduar/boss_assembly_of_iron.cpp

```diff
@@ -15,751 +15,879 @@
-    SAY_BRUNDIR_SPECIAL                         = 2,
-    SAY_BRUNDIR_FLIGHT                          = 3,
-    SAY_BRUNDIR_DEATH                           = 4,
-    SAY_BRUNDIR_ENCOUNTER_DEFEATED              = 5,
-    SAY_BRUNDIR_BERSERK                         = 6,
-    EMOTE_BRUNDIR_OVERLOAD                      = 7
+    ACTION_SUPERCHARGE = 1,
+    ACTION_ADD_CHARGE,
+    ACTION_BERSERK_TRIGGERED
 };
 
-enum Misc
+enum Texts
 {
-    NPC_WORLD_TRIGGER                            = 22515,
+    // Steelbreaker
+    SAY_STEELBREAKER_AGGRO              = 0,
+    SAY_STEELBREAKER_SLAY               = 1,
+    SAY_STEELBREAKER_POWER              = 2,
+    SAY_STEELBREAKER_DEATH              = 3,
+    SAY_STEELBREAKER_ENCOUNTER_DEFEATED = 4,
+    SAY_STEELBREAKER_BERSERK            = 5,
+
+    // Runemaster Molgeim
+    SAY_MOLGEIM_AGGRO                   = 0,
+    SAY_MOLGEIM_SLAY                    = 1,
+    SAY_MOLGEIM_RUNE_DEATH              = 2,
+    SAY_MOLGEIM_SUMMON                  = 3,
+    SAY_MOLGEIM_DEATH                   = 4,
+    SAY_MOLGEIM_ENCOUNTER_DEFEATED      = 5,
+    SAY_MOLGEIM_BERSERK                 = 6,
 
-    DATA_PHASE_3                                 = 1
+    // Stormcaller Brundir
+    SAY_BRUNDIR_AGGRO                   = 0,
+    SAY_BRUNDIR_SLAY                    = 1,
+    SAY_BRUNDIR_SPECIAL                 = 2,
+    SAY_BRUNDIR_FLIGHT                  = 3,
+    SAY_BRUNDIR_DEATH                   = 4,
+    SAY_BRUNDIR_ENCOUNTER_DEFEATED      = 5,
+    SAY_BRUNDIR_BERSERK                 = 6,
+    EMOTE_BRUNDIR_OVERLOAD              = 7
 };
 
-class boss_steelbreaker : public CreatureScript
+enum MovePoints
 {
-    public:
-        boss_steelbreaker() : CreatureScript("boss_steelbreaker") { }
+    POINT_ID_MOVE_INTRO = 1,
+    POINT_ID_FLIGHT_START,
+    POINT_ID_LAND
+};
 
-        struct boss_steelbreakerAI : public BossAI
-        {
-            boss_steelbreakerAI(Creature* creature) : BossAI(creature, BOSS_ASSEMBLY_OF_IRON)
-            {
-                Initialize();
-            }
+enum Data
+{
+    DATA_PHASE_3 = 1
+};
 
-            void Initialize()
-            {
-                phase = 0;
-            }
+std::array<Position, 8> BrundirIntroWaypoints =
+{
+    Position(1563.3f,  120.467f, 427.3189f),
+    Position(1567.86f, 129.128f, 427.3189f),
+    Position(1567.87f, 111.979f, 427.3189f),
+    Position(1576.02f, 107.71f, 427.3189f),
+    Position(1576.19f, 133.968f, 427.3189f),
+    Position(1584.52f, 111.742f, 427.3189f),
+    Position(1584.58f, 129.175f, 427.3189f),
+    Position(1588.82f, 120.42f, 427.3189f)
+};
 
-            uint32 phase;
+CircleBoundary const AssemblyBoundary({ 1587.2f, 121.0f }, 90.0f);
+CreatureBoundary const ArenaBoundaries = { &AssemblyBoundary };
 
-            void Reset() override
-            {
-                _Reset();
-                Initialize();
-                me->RemoveAllAuras();
-            }
+namespace EncounterHelper
```

Comment:
change this namespace name into something specific, like AOIEncounterHelper or w/e
not being nitpicky, this can actually cause linking errors if somewhere else a namespace with the same name is used

for good measure, change the enums in the same way too

Reply:
I think we don't need this for boss encounter files because we do not include anything special in here. I would be totally on board with that when we would have these things in header files but this one here is isolated

Reply:
sadly it doesnt matter if the other file is inlcuded in this or not, its not about that